### PR TITLE
improved node config structure and docs

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: pkg/apis/core/v1alpha1/node_types.go [hunk 1]
```diff
@@ -77,7 +77,15 @@ type NodeConfigSpecGuestClusterAPI struct {
 }
 
 type NodeConfigSpecGuestNode struct {
-	Name string `json:"name" yaml:"name"`
+	// ID is the identifier of the guest cluster's master and worker nodes. In
+	// Kubernetes/Kubectl they are represented as node names. We manage them in an
+	// abstracted way because of provider specific differences.
+	//
+	//     AWS: EC2 instance DNS.
```
Comment (tuommaki) [inline pkg/apis/core/v1alpha1/node_types.go]: I'm curious, why instance DNS instead of instance ID?

File: pkg/apis/core/v1alpha1/node_types.go [hunk 2]
```diff
@@ -77,7 +77,15 @@ type NodeConfigSpecGuestClusterAPI struct {
...
+	//     Azure: ???.
```
Comment (kopiczko) [inline pkg/apis/core/v1alpha1/node_types.go]: Profit :trollface:

File: pkg/apis/core/v1alpha1/node_types.go (hunk 1 above)
Comment (xh3b4sd) [reply pkg/apis/core/v1alpha1/node_types.go]: You have to ask the AWS pros. See https://gigantic.slack.com/archives/C2MS2VB37/p1518003020000197.

File: pkg/apis/core/v1alpha1/node_types.go (hunk 2 above)
Comment (xh3b4sd) [reply pkg/apis/core/v1alpha1/node_types.go]: I actually wanted to have a real answer here. Do we know this already?

**Comment by xh3b4sd** [comment]
@kopiczko I changed `Name` to `ID` because you said this is better in the thread in Slack. Then you changed your mind and I still think `Name` is totally fine since it is about the Kubernetes node name, which is just a k8s specific thing. So I would like to keep `Name` but go with the godoc here.

**Comment by kopiczko** [comment]
@xh3b4sd I'm sorry for the introduced confusion. I didn't realize this is Kubernetes node name. `Name` makes total sense there. The godoc is very nice still.

**Comment by xh3b4sd** [comment]
I still miss information for Azure though. I want to replace `???` with something valuable. `Profit` was a nice joke but I am still waiting for a real answer from you.

File: pkg/apis/core/v1alpha1/node_types.go (hunk 2 above)
Comment (kopiczko) [reply pkg/apis/core/v1alpha1/node_types.go]: I can't provide any useful information before we implement scaling or https://github.com/giantswarm/azure-operator/issues/119 in Azure.

Comment (xh3b4sd) [reply pkg/apis/core/v1alpha1/node_types.go]: I see. Thanks for the answer. Will go with this then and we update later on when we know more.

**Pull request merged**